Lib/test/test_zoneinfo/test_zoneinfo.py

```diff
@@ -1720,13 +1720,35 @@ def test_env_variable_relative_paths(self):
                 with self.subTest("warning", path_var=path_var):
                     # Note: Per PEP 615 the warning is implementation-defined
                     # behavior, other implementations need not warn.
-                    with self.assertWarns(self.module.InvalidTZPathWarning):
+                    with self.assertWarns(self.module.InvalidTZPathWarning) as w:
                         self.module.reset_tzpath()
+                    self.assertEqual(w.warnings[0].filename, __file__)
 
                 tzpath = self.module.TZPATH
                 with self.subTest("filtered", path_var=path_var):
                     self.assertSequenceEqual(tzpath, expected_paths)
 
+    def test_env_variable_relative_paths_warning_location(self):
+        path_var = "path/to/somewhere"
+
+        with self.python_tzpath_context(path_var):
+            with self.subTest("module-level warning"):
+                with CleanImport("zoneinfo", "zoneinfo._tzpath"):
+                    with self.assertWarns(RuntimeWarning) as w:
+                        import zoneinfo
```

Comment:
```suggestion
                        import zoneinfo
                    InvalidTZPathWarning = zoneinfo.InvalidTZPathWarning
```
And now you can use it in `assertIsInstance()` below.

Reply:
Oh, now I get your idea! 👍 Done.